stakedex_sdk/tests/test_main.rs

```diff
@@ -124,30 +753,37 @@ fn test_swap_via_stake(input_mint: Pubkey, output_mint: Pubkey) {
                                 source_token_account.to_string(),
                                 destination_token_account.to_string(),
                             ],
-                            encoding: Some(UiAccountEncoding::Base64),
+                            encoding: Some(UiAccountEncoding::JsonParsed),
                         }),
                         ..RpcSimulateTransactionConfig::default()
                     },
                 )
                 .unwrap();
-            println!(
-                "1 {}\n2 {}",
-                source_token_account, destination_token_account
-            );
-            println!("Simulate result: {:?}", result.value);
-            println!("{:?}", result.value.accounts.unwrap());
-        }
-    }
-}
 
-#[test]
-fn test_swap_via_stake_stsol_unstakeit() {
-    test_swap_via_stake(stsol::ID, native_mint::ID);
-}
+            let res_accounts = result.value.accounts.unwrap();
+            let res_source_account = res_accounts[0].as_ref().unwrap();
+            let res_destination_account = res_accounts[1].as_ref().unwrap();
 
-#[test]
-fn test_swap_via_stake_jsol_msol() {
-    test_swap_via_stake(jsol::ID, msol::ID);
+            if let (Some(decoded_source_account), Some(decoded_destination_account)) = (
+                res_source_account.decode::<Account>(),
+                res_destination_account.decode::<Account>(),
+            ) {
```

Comment:
```suggestion
            let (decoded_source_account, decoded_destination_account) = (
                res_source_account.decode::<Account>().unwrap(),
                res_destination_account.decode::<Account>().unwrap(),
            );
```
decoding shouldnt fail, so we should error out if it does